### ISSUE TITLE
Support lower frequencies on Nano/Uno (Nano not accurate?)

### ISSUE THREAD

**Issue by Jackster** [maxint-rd/FastPwmPin#1]
Trying to get 1,000Hz out of my Nano. 
Setting the example code to "FastPwmPin::enablePwmPin(3, 1000L, 50);" and my meter (which is not great but) is saying that the PWM signal is 125kHz not 1kHz.

Setting it to 1,000,000 is correct on the meter, so is 100,000. But 10,000 goes to 250,000Hz.

Any ideas?

**Comment by maxint-rd** [comment]
Hi Jackster. Sorry for the late reply. I still haven't found where I can enable notifications for Issues. 

I think you ran into a truncation issue. This library is aimed at generating high frequencies. On the 168 (which is close enough to the 328), the lowest frequency is about 31KHz. To support lower frequencies a different timer (the 16-bit Timer1) needs to be used. I wrote that code for the ATtiny44A, but not (yet) for the ATmega328/168. When I can I find the time I will look into that. For now I have renamed this issue and I'll leave this issue open until I've tested/improved it.

FYI The resolution and frequency of the actually generated signal depends on the MCU used and the supported timers. The supplied parameters may be truncated during calculations. The precision of the generated signal depends on this integer truncation.

**Comment by maxint-rd** [comment]
It took some time, but earlier today I uploaded a new version of the library. It now supports generating lower frequencies; as low as 1 Hz on supported pins/MCUs. By using the 16-bit Timer 1 of the ATmega processor and the prescaler, the frequency can be set lower and the precision is higher. See the [readme](https://github.com/maxint-rd/FastPwmPin/blob/master/README.md) for test results. I also added an example that demonstrates this.
